src/FrameAnnotations.cpp

```diff
@@ -0,0 +1,942 @@
+			RE::VTABLE_BSImagespaceShaderISUnderwaterMask[0]);
+		stl::write_vfunc<0xC, BSImagespaceShader_Dispatch<RE::ImageSpaceManager::ISWaterFlow>>(
+			RE::VTABLE_BSImagespaceShaderWaterFlow[0]);
+
+		stl::write_vfunc<0x2A, BSShaderAccumulator_FinishAccumulatingDispatch>(
+			RE::VTABLE_BSShaderAccumulator[0]);
+
+		stl::write_vfunc<0x35, BSCubeMapCamera_RenderCubemap>(RE::VTABLE_BSCubeMapCamera[0]);
+
+		stl::write_vfunc<0xA, BSShadowDirectionalLight_RenderShadowmaps>(
+			RE::VTABLE_BSShadowDirectionalLight[0]);
+		stl::write_vfunc<0xA, BSShadowFrustumLight_RenderShadowmaps>(
+			RE::VTABLE_BSShadowFrustumLight[0]);
+		stl::write_vfunc<0xA, BSShadowParabolicLight_RenderShadowmaps>(
+			RE::VTABLE_BSShadowParabolicLight[0]);
+
+		*(uintptr_t*)&ptr_BSBatchRenderer_RenderBatches = Detours::X64::DetourFunction(REL::RelocationID(100852, 107642).address(), (uintptr_t)&hk_BSBatchRenderer_RenderBatches);
+		*(uintptr_t*)&ptr_Main_RenderDepth = Detours::X64::DetourFunction(REL::RelocationID(100421, 107139).address(), (uintptr_t)&hk_Main_RenderDepth);
+		*(uintptr_t*)&ptr_Main_RenderShadowmasks = Detours::X64::DetourFunction(REL::RelocationID(100422, 107140).address(), (uintptr_t)&hk_Main_RenderShadowmasks);
+		*(uintptr_t*)&ptr_Main_RenderWorld = Detours::X64::DetourFunction(REL::RelocationID(100424, 107142).address(), (uintptr_t)&hk_Main_RenderWorld);
+		*(uintptr_t*)&ptr_Main_RenderFirstPersonView = Detours::X64::DetourFunction(REL::RelocationID(100411, 107129).address(), (uintptr_t)&hk_Main_RenderFirstPersonView);
+		*(uintptr_t*)&ptr_Main_RenderPlayerView = Detours::X64::DetourFunction(REL::RelocationID(35560, 36559).address(), (uintptr_t)&hk_Main_RenderPlayerView);
+		*(uintptr_t*)&ptr_Main_RenderWaterEffects = Detours::X64::DetourFunction(REL::RelocationID(35561, 36560).address(), (uintptr_t)&hk_Main_RenderWaterEffects);
+		*(uintptr_t*)&ptr_BSShaderAccumulator_RenderBatches = Detours::X64::DetourFunction(REL::RelocationID(99963, 106609).address(), (uintptr_t)&hk_BSShaderAccumulator_RenderBatches);
+		*(uintptr_t*)&ptr_BSShaderAccumulator_RenderPersistentPassList = Detours::X64::DetourFunction(REL::RelocationID(100840, 107630).address(), (uintptr_t)&hk_BSShaderAccumulator_RenderPersistentPassList);
+		*(uintptr_t*)&ptr_BSShaderAccumulator_RenderEffects = Detours::X64::DetourFunction(REL::RelocationID(99940, 106585).address(), (uintptr_t)&hk_BSShaderAccumulator_RenderEffects);
+		*(uintptr_t*)&ptr_VolumetricLightingDescriptor_Render = Detours::X64::DetourFunction(REL::RelocationID(100306, 107023).address(), (uintptr_t)&hk_VolumetricLightingDescriptor_Render);
+	}
+
+	void OnDataLoaded()
+	{
+		auto renderer = RE::BSGraphics::Renderer::GetSingleton();
+
+		for (size_t renderTargetIndex = 0;
+			 renderTargetIndex < (!REL::Module::IsVR() ? RE::RENDER_TARGETS::kTOTAL : RE::RENDER_TARGETS::kVRTOTAL); ++renderTargetIndex) {
+			const auto renderTargetName = magic_enum::enum_name(
```

Comment:
Add prefix to the privatedata setters maybe? The enum names are sometimes the same in `RENDER_TARGET_DEPTHSTENCIL` and `RENDER_TARGET`

For example `kMAIN, kMAIN_COPY`

Reply:
Can you give an example of how it would differ?

Reply:
I mean seems like a good addition, but not sure if it has to gate this PR. If you want to submit a PR, then I can also just merge this in. 

Only concern may be whether the name gets too long. But perhaps an abbreviation can help.

Reply:
I think this works for dev. We can make future pr for the prefixes.